File: server-side/src/main/java/com/acm/server/model/dto/ArchiveDto.java
package com.acm.server.model.dto;

import lombok.Data;

import java.util.Date;
import java.util.List;

/**
 * Archive DTO
 * @see com.acm.server.domain.Archive
 * @see com.acm.server.model.dto.PictureDto
 * @see com.acm.server.model.dto.FileDto
 *
 * @author Farid Masjedi
 */
@Data
public class ArchiveDto {
    private Long id;
    private Date date;
    private FileDto questions;
    private FileDto rankings;
    private List<PictureDto> eventDayPictures;
    private PictureDto poster;
}

File: client-side/src/app/shared/interfaces/DTO/archive.dto.ts
import {FileDto} from "./file.dto";
import {PictureDto} from "./picture.dto";

export class ArchiveDto {
  id!: number;
  date!: Date;
  questions!: FileDto;
  rankings!: FileDto;
  eventDayPictures!: PictureDto[];
  poster!: PictureDto;
}
